### ISSUE TITLE
support ln command on OSX

### ISSUE THREAD

**Issue by simonmichael** [stackbuilders/hapistrano#63]
This looks pretty good, thanks!

Testing a deployment on OSX failed, /bin/ln doesn't support -T:
```
$ ln -svT /repos/demo.hledger.org/releases/20170329171058/ /repos/demo.hledger.org/current_tmp
ln: illegal option -- T
usage: ln [-Ffhinsv] source_file [target_file]
       ln [-Ffhinsv] source_file ... target_dir
       link source_file target_file
```

**Comment by jsl** [comment]
Thanks @simonmichael for bringing this to our attention. It should work on OS X. We'll take a look!

**Comment by patrixl** [comment]
This affects FreeBSD also (not surprisingly, both are BSD userland), and not just ln, but also mv.

**Comment by juanpaucar** [comment]
@simonmichael @patrixl @jsl Hi everyone, I know this issue has been around for a while but I wanted your opinion on this. Do you think it'd be better to enable another key on the YAML file to tell the *hapistrano* that the target deployment system won't be a linux system and instruct it to use the command versions without unsupported options instead?
 
It would be something like the normal configuration plus another key like `linux` or `bsd` and set accordingly
```yaml
deploy_path: '/var/projects/my-project'
host: myserver.com
port: 2222
repo: 'https://github.com/stackbuilders/hapistrano.git'
revision: origin/master
build_script:
  - stack setup
  - stack build
restart_command: systemd restart my-app-service

linux: false
```

**Comment by patrixl** [comment]
@juanpaucar Hi! Yes a key in the YAML file would probably make everyone's life simpler, I'm fine with that.

**Comment by juanpaucar** [comment]
Sorry for the very late solving of this issue. This has been solved in #86 and it's available starting from hapistrano 0.3.5.0
Thanks @simonmichael and @patrixl for reporting the issue
